tests/gocase/unit/kmetadata/kmetadata_test.go

```diff
@@ -0,0 +1,276 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package kmetadata
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/redis/go-redis/v9"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/apache/kvrocks/tests/gocase/util"
+)
+
+type kMetadataResponse struct {
+	ttl     int64  `redis:"ttl"`
+	size    int64  `redis:"size"`
+	ktype   string `redis:"type"`
+	flags   int64  `redis:"flags"`
+	version int64  `redis:"version"`
+}
+
+func ExtractKMetadataResponse(result interface{}) (*kMetadataResponse, error) {
+	// Check if result is a map
+	resultMap, ok := result.(map[interface{}]interface{})
+	if !ok {
+		return nil, fmt.Errorf("expected map[interface{}]interface{}, got %T", result)
+	}
+
+	response := &kMetadataResponse{}
+
+	// Extract TTL field
+	if val, ok := resultMap["ttl"]; ok {
+		switch v := val.(type) {
+		case int64:
+			response.ttl = v
+		case int:
+			response.ttl = int64(v)
+		case float64:
+			response.ttl = int64(v)
+		default:
+			return nil, fmt.Errorf("ttl is not a number, got %T", val)
+		}
+	}
```

Comment:
Can we extract this type's checker using a dedicated function to avoid duplicate codes?